lib/trino-filesystem/src/main/java/io/trino/filesystem/TrinoOutputFile.java

```diff
@@ -35,11 +35,20 @@ default OutputStream createOrOverwrite()
         return createOrOverwrite(newSimpleAggregatedMemoryContext());
     }
 
+    default OutputStream createExclusive()
```

Comment:
@electrum Besides the `IOException`, we should throw another exception to be able to figure out whether we're dealing with a logical conflict or an actual IOException.
This kind of information could be used in `trino-delta-lake` to figure out whether we are having a transaction conflict which can be eventually solved through retries or we have an IO exception which fails the transaction.